File: LATIHAN4DPBO2023/Python/Mahasiswa.py
from Sivitas import Sivitas

#inisialisasi class Mahasiswa yang merupakan child dari class Sivitas
class Mahasiswa(Sivitas):
	# atribut
	__nim = ""
	__faculty = ""
	__major = ""

	# constructor
	def __init__(self, nik = "", name = "", gender = "", univ = "", email = "", nim = "", faculty = "", major = ""):
		super().__init__(nik, name, gender, univ, email)
		self.__nim = nim
		self.__faculty = faculty
		self.__major = major

	# kumpulan method setter dan getter untuk tiap atribut
	def setNim(self, nim):
		self.__nim = nim

	def getNim(self):
		return self.__nim

	def setFaculty(self, faculty):
		self.__faculty = faculty

	def getFaculty(self):
		return self.__faculty

	def setMajor(self, major):
		self.__major = major

	def getMajor(self):
		return self.__major

File: LATIHAN4DPBO2023/CPP/Mahasiswa.cpp
/*Saya Mochamad Khaairi NIM 2106416 mengerjakan Latihan 4
dalam mata kuliah Desain dan Pemrograman Berorientasi Objek 
untuk keberkahanNya maka saya tidak melakukan kecurangan 
seperti yang telah dispesifikasikan. Aamiin.*/

//include file header
#include "header.hh"

//inisialisasi constructor
Mahasiswa::Mahasiswa() : Sivitas()
{
	nim = "";
	faculty = "";
	major = "";
}

//inisialisasi constructor dengan parameter (menggunakan supper dengan class Sivitas)
Mahasiswa::Mahasiswa(string nik, string name, string gender, string univ, string email, string nim, string faculty, string major) : Sivitas(nik, name, gender, univ, email)
{
	this->nim = nim;
	this->faculty = faculty;
	this->major = major;
}

//inisialisasi method setter dan getter
void Mahasiswa::setNim(string nim)
{
	this->nim = nim;
}

string Mahasiswa::getNim()
{
	return this->nim;
}

void Mahasiswa::setFaculty(string faculty)
{
	this->faculty = faculty;
}

string Mahasiswa::getFaculty()
{
	return this->faculty;
}

void Mahasiswa::setMajor(string major)
{
	this->major = major;
}

string Mahasiswa::getMajor()
{
	return this->major;
}

//inisialisasi destructor
Mahasiswa::~Mahasiswa()
{
}